### PR TITLE
[rtsan] Add I/O multiplexing interceptors

### PR DESCRIPTION
**Pull request by cjappl**
Intercepts in the family of `poll`, `select` and modern equivalents `epoll` (linux only) and `kqueue` bsd family only.

These calls mirror the names of the system calls they call, which have been verified on mac at least (e.g. kevent calls the system call kevent).

### CODE REVIEW

File: compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp [hunk 1]
```diff
@@ -614,6 +615,104 @@ INTERCEPTOR(int, shutdown, int socket, int how) {
...
+
+INTERCEPTOR(int, poll, struct pollfd *fds, nfds_t nfds, int timeout) {
+  __rtsan_notify_intercepted_call("poll");
+  return REAL(poll)(fds, nfds, timeout);
+}
+
+#if !SANITIZER_APPLE
+// FIXME: This should work on all unix systems, even Mac, but currently
+// it is showing some weird error while linking
+// error: declaration of 'select' has a different language linkage
```
Comment (cjappl) [inline compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp:628]: Drawing folks attention here, this is very strange and I haven't seen it for an interceptor before

**Comment by cjappl** [comment]
Weekly ping of reviewers @davidtrevelyan @fmayer . Thanks :)

File: compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp (hunk 1 above)
Comment (davidtrevelyan) [reply compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp:628]: @cjappl  There might be some clues comparing the declarations of `select` on Darwin and other unix OSes. Did you already take a look? I can investigate when I have some time (probably not for a couple of weeks, though, unfortunately)

Comment (cjappl) [reply compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp:628]: I took a look but was not able to find anything interesting. Definitely let me know if you see something I didn't see, I'd be curious!

**Pull request merged**